### PR TITLE
quint/grammar: add support for optional leading hashbang line

### PR DESCRIPTION
**Pull request by lucab**
This updates Quint grammar to support a leading hashbang (`#!`) line in spec files.
It allows using a spec file as a self-running script on Unix systems, as shown in `SuperSpec.qnt`.

- [x] Tests added for any new code
- [x] Entries added to the respective `CHANGELOG.md` for any new functionality

Closes https://github.com/informalsystems/quint/issues/1500

### CODE REVIEW

**Comment by lucab** [comment]
/cc @bugarela for review.

I decided to follow JS approach and only carve a niche in the grammar for the first line, see https://github.com/tc39/proposal-hashbang for details.

File: quint/src/generated/Quint.g4 [hunk 1]
```diff
@@ -313,6 +314,7 @@ CAP_ID : ([A-Z][a-zA-Z0-9_]*|[_][a-zA-Z0-9_]+) ;
 DOCCOMMENT : '///' .*? '\n';
 
 // comments and whitespaces
+HASHBANG_LINE   :   '#!' .*? '\n'   -> skip ; // only used for the first line of a file
```
[the remark this answers is not in the document]
Comment (lucab) [reply quint/src/generated/Quint.g4]: Apologies, I'm not overly familiar with ANTLR grammars so I didn't realize that `skip` had such a side-effect.
I did the changes you suggest and also added a few unit tests to ensure that the hashbang rule only applies to the first line of the file.

**Comment by lucab** [comment]
Fixed grammar rules and rebased to latest `main`. Also added an integration test at the bottom of `cli-tests.md` to run the typechecking on `SuperSpec.qnt` via the hashbang on Unix.

**Review by bugarela — APPROVED** [review]
Looks great! Thanks again

**Pull request merged**